app/controllers/api_keys_controller.rb

```diff
@@ -26,10 +27,18 @@ def create
       flash[:error] = @api_key.errors.full_messages.to_sentence
       render :new
     end
+  rescue ActiveRecord::RecordNotFound
+    flash[:error] = t(".invalid_gem")
+    @api_key = current_user.api_keys.build(api_key_params.merge(rubygem_id: nil))
+    render :new
   end
 
   def edit
     @api_key = current_user.api_keys.find(params.require(:id))
+    return unless @api_key.soft_deleted?
```

Comment:
Do we need to add error message and forbidden status here?

Reply:
Oh, ignore my previous comment. msg and status is needed "unless" key is soft deleted.